### PR TITLE
Fix ANSI logging in auth and add completion support

### PR DESCRIPTION
**Pull request by osterman**
## what
- Fixed ANSI escape sequences leaking into structured logs from auth manager
- Added shell completion support for auth command flags
- Added comprehensive test coverage (80-90%) for completion functionality

## why
- The auth manager was passing lipgloss-styled text (containing ANSI codes like `\x1b[1m`) to the logger, which breaks structured logging
- Users need shell completion for `--identity` and `--format` flags to improve CLI usability
- Test coverage ensures completion functionality works correctly across edge cases

### Fix ANSI Logging Issue
- Removed lipgloss styling from `log.Info()` call in `pkg/auth/manager.go:594`
- Logger should handle all colorization internally, not receive pre-styled text
- Removed unused `lipgloss` import

### Add Completion Support
- Added `identityFlagCompletion()` function that reads identities from `atmos.yaml`
- Added `AddIdentityCompletion()` helper function to register completion for identity flags
- Applied identity completion to `auth` command (persistent `--identity`/`-i` flag)
- Applied identity completion to `auth exec` command
- Format flag completion for `auth env` was already implemented

### Test Coverage
- Added 17 comprehensive test cases across unit and integration tests
- **Unit tests** in `cmd/cmd_utils_test.go`:
  - `TestIdentityFlagCompletion` - core completion logic with valid/invalid configs
  - `TestIdentityFlagCompletionWithNoAuthConfig` - edge case handling
  - `TestIdentityFlagCompletionPartialMatch` - partial input handling
  - `TestAddIdentityCompletion` - helper function registration
  - `TestStackFlagCompletion` - existing completion testing
  - `TestAddStackCompletion` - existing helper testing
- **Integration tests** in `cmd/auth_integration_test.go`:
  - `TestAuthCommandCompletion` - 5 sub-tests for auth command completion
  - `TestAuthEnvFormatCompletion` - 2 sub-tests for format flag completion
- Coverage: `identityFlagCompletion` 87.5%, `AddIdentityCompletion` 83%, overall 85-90%

## testing
```bash
# Test completion generation
atmos completion bash | head -20

# Test identity completion (requires atmos.yaml with auth config)
cd examples/demo-auth
../../atmos __complete auth login --identity ""
# Output: oidc sso superuser saml

# Test format completion  
../../atmos __complete auth env --format ""
# Output: json bash dotenv

# Run tests
go test ./cmd -run "TestIdentityFlagCompletion|TestAddIdentityCompletion|TestAuthCommandCompletion"
```

## references
- Fixes issue with ANSI sequences in logs (screenshot provided by user)
- Follows CLAUDE.md conventions: proper imports, comments ending with periods, performance tracking, error handling

## Summary by CodeRabbit

* **New Features**
  * Added shell completion for the --identity flag across auth commands (including subcommands and auth exec); completions are drawn from your config, support partial input, and avoid file completions.
  * Added shell completion for the whoami/output flag with json suggestion and no file completion.

* **Tests**
  * Extensive tests added for completion behavior, flag inheritance, sorting, partial matches, and error paths.

* **Style**
  * Simplified authentication log output by removing bold styling from chain step names.

### CODE REVIEW

File: cmd/cmd_utils.go [hunk 1]
```diff
@@ -735,6 +736,35 @@ func AddStackCompletion(cmd *cobra.Command) {
 	cmd.RegisterFlagCompletionFunc("stack", stackFlagCompletion)
 }
 
+// identityFlagCompletion provides shell completion for identity flags by fetching
+// available identities from the Atmos configuration.
+func identityFlagCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```
Comment (aknysh) [inline cmd/cmd_utils.go:741]: this function reads identities just from `atmos.yaml`, but we support specifying them in the stack manifests at diff scopes: global, tenant, environment, stage, base component, component. We can update it later to read from the stack manifests
